### DISCUSSION TITLE
Need help displaying my code on a Forumcommunity

### DISCUSSION THREAD

**Discussion by magple** [community/community#132549] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hey, so I'm just a newbie at coding, but I have finished a template for a roleplaying story and I want to upload and see the file on a Forumcommunity website. I have seen other people link their github to their post and it shows their template perfectly, however, when I try to link mine, it doesn't show anything. 

Forumcommunity usually just lets you work with basic html and inline css, so that's why I decided to first upload to github and then link it to the website since I've seen other projects run smoothly this way. 

One way to see how other people have created their code is by clicking "citation" on their post and it shows all their code. This person's code only said (FILE:[URL:*link*]*link again*[/URL]). I tried replicating this method, and nothing happened.

Please, any help would be much appreciated. I have been looking everywhere and can't find anything that answers my issue.

**Comment by nikolozkoiava** [comment]
Go to your HTML file on GitHub.
Click on the "Raw" button to view the plain HTML version.
Copy the URL from the address bar.

Let's say your GitHub file URL is
https://github.com/your-username/your-repo/blob/main/template.html

The raw URL would be
https://raw.githubusercontent.com/your-username/your-repo/main/template.html

Using the iframe method, your code is in Forumcommunity.

<iframe src="https://raw.githubusercontent.com/your-username/your-repo/main/template.html></iframe>

**Comment by danialtavakoli** [comment]
It sounds like you're encountering issues with displaying your HTML and CSS template properly on a Forumcommunity website after linking it from GitHub. Here are some steps and considerations that might help you troubleshoot and resolve the issue:

### 1. **GitHub Link Setup**
   - Make sure your HTML file (let's say it's named `template.html`) and any associated CSS (e.g., `styles.css`) are correctly uploaded to GitHub.
   - Verify that the file structure on GitHub matches what you expect. For example, if your files are in a folder named `template`, your GitHub link should reflect this structure (e.g., `https://github.com/your-username/repository-name/tree/main/template/template.html`).

### 2. **Linking on Forumcommunity**
   - Forumcommunity typically allows basic HTML and inline CSS, but it's important to ensure that your HTML file is structured correctly for embedding.
   - When you post on Forumcommunity, use the appropriate HTML tags to embed your GitHub-hosted content. For instance:
     ```html
     <iframe src="https://github.com/your-username/repository-name/blob/main/template/template.html"></iframe>
     ```
     Replace `your-username` and `repository-name` with your GitHub username and repository name respectively.

### 3. **Check Embedding Options**
   - Some forums might have restrictions or specific formatting requirements for embedding external content like GitHub files. Make sure you're using the correct embedding syntax supported by Forumcommunity.

### 4. **Direct Linking**
   - If embedding via iframe doesn't work or isn't allowed, you might try providing a direct link to your GitHub repository or to the raw HTML file. Users can then click the link to view the template directly on GitHub.

### 5. **Permissions and Visibility**
   - Ensure that your GitHub repository and files are set to be publicly accessible. Private repositories or files won't be viewable unless users have the appropriate permissions.

### 6. **Testing and Troubleshooting**
   - Test your links and embedding on different browsers and devices to see if the issue persists across different platforms.
   - Review any error messages or console logs in your browser’s developer tools when attempting to view the embedded content.

### Example of Correct Usage
   - Suppose your GitHub repository URL is `https://github.com/your-username/repository-name`.
   - Your HTML file `template.html` is located directly in the repository.
   - You would link to it like this in HTML:
     ```html
     <iframe src="https://github.com/your-username/repository-name/blob/main/template.html"></iframe>
     ```
   - Adjust the URL structure (`blob/main/`) according to where your files are located in the GitHub repository.

By following these steps and ensuring that your GitHub repository and Forumcommunity post settings are configured correctly, you should be able to successfully link and display your HTML and CSS template. If you encounter specific error messages or behaviors, those details could provide more clues for further troubleshooting.
